packages/react-compose/src/utils.ts

```diff
@@ -24,13 +24,39 @@ export const defaultComposeOptions: ComposePreparedOptions = {
 
   handledProps: [] as never[],
   overrideStyles: false,
+  slots: {},
+  mapPropsToSlotPropsChain: [],
+  resolveSlotProps: () => ({}),
 };
 
 export function mergeComposeOptions(
   input: Input,
   inputOptions: ComposeOptions,
   parentOptions: ComposePreparedOptions = defaultComposeOptions,
 ): ComposePreparedOptions {
+  const mapPropsToSlotPropsChain = inputOptions.mapPropsToSlotProps
+    ? [...parentOptions.mapPropsToSlotPropsChain, inputOptions.mapPropsToSlotProps]
+    : parentOptions.mapPropsToSlotPropsChain;
+
+  const resolveSlotProps = <P = {}>(props: P) =>
+    mapPropsToSlotPropsChain.reduce<Record<string, object>>((acc, definition) => {
+      const nextProps = definition(props) || {};
```

Comment:
nit:

```suggestion
      const nextProps =  {...definition(props)};
```

if `definition(props)` returns `undefined` the result of would be just `{}`. Same could be used for `slots` on line 73.

Does it make sense?